crates/sui-types/src/effects/object_change.rs

```diff
@@ -66,6 +67,37 @@ pub enum ObjectIn {
     Exist((VersionDigest, Owner)),
 }
 
+#[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize)]
+pub enum AccmulatorOperation {
+    /// Merge the value into the accumulator.
+    Merge,
+    /// Split the value from the accumulator.
+    Split,
+}
+
+#[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize)]
+pub enum CachedAccumulatorValue {
+    NotCached,
+    Integer(u128),
+    IntegerTuple(u128, u128),
+}
+
+#[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize)]
+pub struct AccumulatorWriteV1 {
+    /// The recipient of the accumulator.
+    pub recipient: SuiAddress,
+    /// The type of the accumulator.
+    pub accumulator_type: TypeTag,
+    /// The operation to be applied to the accumulator.
+    pub operation: AccmulatorOperation,
+    /// The raw value to be applied to the accumulator.
+    /// It can be interpreted based on the type of the accumulator.
+    pub raw_value: Vec<u8>,
+    /// The cached deserialized value of the operation.
+    /// This is an optimization to avoid deserializing the value multiple times.
+    pub cached_value: CachedAccumulatorValue,
```

Comment:
You're likely not going to want this duplicated, or you're going to want to skip ser/de on one of these fields. As it stands right now you have this data duplicated

Reply:
I don't have a very strong opinion on this. @mystenmark believes that we should 1) commit to the raw value in effects for safety 2) also include cached value to avoid redundant work.

Reply:
I think a good balance would be to make the raw value another variant of AccumulatorValue, so that for the simpler system cases, we always use deserialized values, but in the future if we ever support custom accumulators, they go to the raw value so effects don't have to worry about its semantics.

Reply:
I don't know the intended use case here but i would actually expect the opposite of that. In that the actual serialized data format just has the raw value as bytes but that we also have a deserialized form of that data that can be done trivially by looking at the type and the raw data. The deserialized form is not part of the serialized format of this type but is just an ephemeral, in-memory thing